src/Microsoft.Identity.Client/Platforms/Android/TokenCacheAccessor.cs

```diff
@@ -54,50 +52,92 @@ public TokenCachePlugin()
                 throw new MsalException("Fail to create SharedPreference");
             }
         }
-        
-        public TokenCachePlugin(RequestContext requestContext)
-        {
-            this.requestContext = requestContext;
-        }
 
-        public ICollection<string> GetAllAccessTokens()
+        public TokenCacheAccessor(RequestContext requestContext) : this()
         {
-            return _accessTokenSharedPreference.All.Values as ICollection<string>;
+            _requestContext = requestContext;
         }
 
-        public ICollection<string> AllRefreshTokens()
+        public void SaveAccessToken(AccessTokenCacheItem accessTokenItem)
         {
-            return _refreshTokenSharedPreference.All.Values as ICollection<string>;
+            ISharedPreferencesEditor editor = _accessTokenSharedPreference.Edit();
+            editor.PutString(accessTokenItem.GetTokenCacheKey().ToString(), JsonHelper.SerializeToJson(accessTokenItem));
+            editor.Apply();
         }
 
-        public void SaveAccessToken(string cacheKey, string accessTokenItem)
+        public void SaveRefreshToken(RefreshTokenCacheItem refreshTokenItem)
```

Comment:
hm, why you change it to tokenCacheItem? we can do serialize/deserialize in the PCL, only thing different across the platforms is the actual storage location, the JSON format should remain the same.